### PR TITLE
add type argument to categorical and categorical!

### LINKED ISSUE #1282: Categorical! to include Char
The current "default" behavior is to make `AbstractVector{T} where T <: AbstractString` a `CategoricalArrays.CategoricalVector` and more specifically `CategoricalArrays.CategoricalVector{T} where {T <: CategoricalArrays.CategoricalString, ...}`

However, I believe that a better default would be to include columns that are `eltype(col) <: Char` as well.

### PR DESCRIPTION
**Pull request by bkamins**
Fixes #1282

### CODE REVIEW

**Comment by bkamins** [comment]
CI fails only do to coverage on deprecated.jl which is not touched by this PR.

**Comment by nalimilan** [comment]
I wonder whether it wouldn't be better to use a positional argument for this. Not sure why, maybe because types are often used that way. What do you think?

**Comment by bkamins** [comment]
Makes sense, and I would say I know why 😄. You either pass a list of columns as a second argument **or** you pass the type as column selector. I have just pushed this fix.

File: src/dataframe/dataframe.jl [hunk 1]
```diff
@@ -972,11 +972,15 @@ disallowmissing!(df::DataFrame, cols::Colon=:) =
                  compress::Bool=false)
     categorical!(df::DataFrame, cnames::Union{Regex, Not};
                  compress::Bool=false)
-    categorical!(df::DataFrame; compress::Bool=false)
+    categorical!(df::DataFrame, cols::Type=Union{AbstractString, Missing};
```
Comment (nalimilan) [inline src/dataframe/dataframe.jl]: Maybe put this one first since it's the simplest method?

Comment (bkamins) [reply src/dataframe/dataframe.jl]: done

**Pull request merged**